caas/kubernetes/provider/k8s.go

```diff
@@ -1192,6 +1190,22 @@ func (k *kubernetesClient) EnsureService(
 	return nil
 }
 
+func validateDeploymentType(t caas.DeploymentType, params *caas.ServiceParams, workloadSpec *workloadSpec) error {
+	if t != caas.DeploymentStateful {
+		if workloadSpec.Service != nil && workloadSpec.Service.ScalePolicy != "" {
+			return errors.NewNotValid(nil, fmt.Sprintf("ScalePolicy is only supported for %s applications", caas.DeploymentStateful))
+		}
+	}
+
+	if t != caas.DeploymentStateful && len(params.Filesystems) > 0 {
```

Comment:
I'm not sure we want to disallow all storage. tmpfs and rootfs could be allowed as these storage types are local to the node of which the charm runs. We can consider just disallowing "kubernetes" storage

Reply:
As discussed, we are going to reject this for now because we don't support it yet.
We will remove this validation once we support it in the future.